server/src/main/java/org/elasticsearch/index/mapper/NumberFieldMapper.java

```diff
@@ -248,6 +270,13 @@ public Float parse(XContentParser parser, boolean coerce) throws IOException {
                 return parsed;
             }
 
+            @Override
+            public BiFunction<Script, ScriptService, NumberMapperScript> compiler(String fieldName) {
```

Comment:
I wish that there were less functions and indirections: it is hard to follow the flow. I guess this has to do with the script service being available only from ParseContext, when mappings are being parsed. And it does need to be on the mapper because the returned type changes for each impl? Is there a chance that we can have an abstraction around a compiled script, with a generic type, that can be used everywhere, and the compiler method does not need to be on every mapper that allows to index calculated values? I see that down the line, a T type is used in CompiledScriptParameter anyways, which throws me off.

Reply:
Yeah, I think we might be able to combine NumberMapperScript and CompiledScriptParameter, I'll give that a go.